### DISCUSSION TITLE
Replace the Prerequisitestables

### DISCUSSION THREAD

**Discussion by nschonni** [mdn/mdn-community#249] [Browser compatibility data]
There are still around 220 leftover from the HTML conversion and represent most of the inline tables remaining in the repo
https://github.com/search?q=%3Cth+scope%3D%22row%22%3EPrerequisites%3A%3C%2Fth%3E+repo%3Amdn%2Fcontent&type=Code&ref=advsearch&l=&l=

Not sure what the proper format for these would be though, or if it could be automated conversion.

**Comment by teoli2003** [comment]
What about transforming these in `<dt>`/`<dl>` lists (with the `- :` Markdown syntax) ?

**Reply by nschonni** [reply]
That might be one approach. I haven't checked many, but here is a sample one
```html
table>
  <tbody>
    <tr>
      <th scope="row">Prerequisites:</th>
      <td>
        Basic computer literacy, a basic understanding of HTML, CSS, and
        JavaScript, an understanding of the
        <a href="/en-US/docs/Learn/Accessibility"
          >previous articles in the course</a
        >.
      </td>
    </tr>
    <tr>
      <th scope="row">Objective:</th>
      <td>To test basic knowledge of accessibility fundamentals.</td>
    </tr>
  </tbody>
</table>
```

Maybe it's even just a plain list like

```md
- **Prerequisites**: Basic computer literacy, a basic understanding of HTML, CSS, and
        JavaScript, an understanding of the [previous articles in the course](/en-US/docs/Learn/Accessibility)
- **Objective**: To test basic knowledge of accessibility fundamentals.
```

**Reply by nschonni** [reply]
Noticing these are all under the `/learn/`, so maybe another approach would be to move it to frontmatter and then it could be tweaked with macros going forward

**Reply by teoli2003** [reply]
```
- : Prerequisites
  - Basic computer literacy.
  - Basic understanding of HTML, CSS, and JavaScript.
  - Understanding of the [previous articles in the course](/en-US/docs/Learn/Accessibility)
- : Objectives
  - Test basic knowledge of accessibility fundamentals.
```

**Reply by nschonni** [reply]
I'm always a bit iffy on using DL/DT, but I think you're probably right. Unfortunately, I thing any automated cleanup could only do a basic DD value.
I think you might have it reversed though, wouldn't it be

```md
- Prerequisites
  - : 
    - Basic computer literacy.
    - Basic understanding of HTML, CSS, and JavaScript.
    - Understanding of the [previous articles in the course](/en-US/docs/Learn/Accessibility)
- Objectives
  - : Test basic knowledge of accessibility fundamentals.
 ```

or the plain way

```md
- Prerequisites
  - : Basic computer literacy, a basic understanding of HTML, CSS, and
        JavaScript, an understanding of the [previous articles in the course](/en-US/docs/Learn/Accessibility)
- Objective
  - : To test basic knowledge of accessibility fundamentals.
```

**Reply by teoli2003** [reply]
What about:

```
## Information

### Prerequisites

- Basic computer literacy.
- Basic understanding of HTML, CSS, and JavaScript.
- Understanding of the [previous articles in the course](/en-US/docs/Learn/Accessibility)

### Objectives

   Test basic knowledge of accessibility fundamentals.
```

This would lead to:
![Capture d’écran 2022-09-03 à 08 09 30](https://user-images.githubusercontent.com/1466293/188258298-367b6e2c-7d45-43ab-a4a1-f30e54b88d77.png)

cc/ @dipikabh that always has good insight on page structure.

**Reply by Josh-Cena** [reply]
FWIW I don't really like it... The good thing about a table is it's visually separated from the main content. I don't need to spend time figuring out where's the boilerplate/metadata and where the main content starts.

**Reply by nschonni** [reply]
You don't like the idea of the metadata, or changing the tables at all? These seem like a "tables for layout" than actual tabular data. The metadata was just spitballing possible ways to get the format flexible

**Reply by hamishwillee** [reply]
@nschonni Well, it's boilerplate. Academically I think it is useful to set expectation, but in practice I don't ever read these, and when authoring I find them painfully repetitive.

@Josh-Cena seems to be saying that since this is the case, having it visually different makes that choice easier for the reader.

I agree with that point and that a table is "better" than DL (and much better than the heading layout) for this purpose.
But IMO DL is _sufficiently_ different from the main text that it is "OK" and losing the HTML is worth the cost of changing. The way it would render in both options shown below. 

- This is what it looks like now:
  ![image](https://user-images.githubusercontent.com/5368500/189247279-2173e929-76ca-4353-8246-548dd7b735f8.png)

- This is what it looks like as a DL:
  ![image](https://user-images.githubusercontent.com/5368500/189247476-368217c4-0736-4d82-aa7d-defbf5f674fa.png)


Just my two bits.

**Reply by hamishwillee** [reply]
PS And I'd be OK with the pre-requisites split up and bulleted too.

**Reply by Josh-Cena** [reply]
I find the "box" to be the important bit. I'm also fine with either DL or table, but I'd like a visually dividing box wrapping the DL so I instantly know which part to jump :)

(And yes, the "prerequisites" are woefully repetitive and uninformative)

**Reply by hamishwillee** [reply]
@Josh-Cena Half a box :-)

![image](https://user-images.githubusercontent.com/5368500/189252614-b0a4557b-f00b-4f7b-af57-f6574b0682a0.png)

**Reply by hamishwillee** [reply]
Actually, probably a quote, or we could add something like like a note syntax to create a highlighted box

![image](https://user-images.githubusercontent.com/5368500/189252893-03d2c3df-d2a9-4919-9077-c82afe99640f.png)

**Reply by Josh-Cena** [reply]
Maybe `> **Callout**`? Looks blank enough. I don't like the idea of using blockquotes simply as a marker since it has a clear semantic.

**Reply by hamishwillee** [reply]
Sure. It's the idea that's important. @nschonni can work out the detail :-) It might be `> **boilerplate**`.

**Reply by Josh-Cena** [reply]
I mean, `> **Callout**` is already a thing so...

**Reply by wbamberg** [reply]
Yes, this is exactly what callouts are for:

<img width="832" alt="Screen Shot 2022-09-12 at 10 24 20 AM" src="https://user-images.githubusercontent.com/432915/189718778-3ba4efbc-8fb5-4c9a-b6ff-ad339d5b6465.png">

What you can't do is have a list immediately inside a `<dt>`, without a text element preceding it:

```
- this

  - :
    - won't
    - work
 ```

```
- but

  - : this
    - works
    - fine
```

I think this might be https://github.com/mdn/yari/issues/4307.

**Reply by hamishwillee** [reply]
So is the proposal that we start converting these now using the format below? Or are we waiting on that yari item?

```md
> **Callout:**
> - Prerequisites
>   - Basic computer literacy.
>   - Basic understanding of HTML, CSS, and JavaScript.
>   - Understanding of the [previous articles in the course](/en-US/docs/Learn/Accessibility)
> - Objectives
>   - Test basic knowledge of accessibility fundamentals.
```

**Reply by wbamberg** [reply]
> So is the proposal that we start converting these now using the format below? 

Speaking for myself, I think:

* this would be fine. I think it's easier to read than the original and rids us of some HTML tables.
* `<ul>` inside `<dd>` is better, but not better enough to be worth waiting for a fix in the Markdown processor.

I agree with @Josh-Cena that the prerequisites are of questionable value, but think that's a separate issue.

**Reply by nschonni** [reply]
It goes against the way that Yari is going, but it could be done with a macro, to kick out the format discussion:
```
{{Prerequisites("Basic computer literacy.", "Basic understanding of HTML, CSS, and JavaScript.", "Understanding of the [previous articles in the course](/en-US/docs/Learn/Accessibility)")}}
{{Objectives("Test basic knowledge of accessibility fundamentals")}}
```
Let a macro decide on the output of the list of parameters.

This solution is likely a lot of work, for not a lot of payoff 😆

**Reply by wbamberg** [reply]
Yeah, and I think it is super-clunky to pass lots of prose into a macro. If the prereqs were just links then I would suggest making them metadata for learn articles, but since they are prose I don't think it's a good plan.

**Reply by hamishwillee** [reply]
IMO we use the [proposed version](https://github.com/mdn/content/discussions/20256#discussioncomment-3630992) or dump the section all together. Anything else is a waste of breath.

> I agree with @Josh-Cena that the prerequisites are of questionable value, but think that's a separate issue.

Let's have the discussion then. I've yet to hear anyone say they think it adds value so could you @wbamberg or maybe @Rumyra take it to OWD discussion?

**Reply by hamishwillee** [reply]
Are we going to progress this?/ Any further discussion?

**Reply by nschonni** [reply]
If this goes forward, should it get spun up on OWD? I realized afterwards, that the discussion on this repo isn't really used.

**Reply by wbamberg** [reply]
> Are we going to progress this?/ Any further discussion?

It is on my list but I didn't get around to it yet :). I'm planning to make a list of all the prerequisites for each page and taking a look to see if they are useful.

**Reply by wbamberg** [reply]
I made a spreadsheet listing all the pages in Learn that contain one of these tables, with the contents of their Prerequisites and Objective cells: https://docs.google.com/spreadsheets/d/1winY_WurPrv1iLPEfx-gy8SI5UiCLDwvIJzbkDBnf3E/edit?usp=sharing.

...the idea being that it could give us an idea of whether they are in general useful or not.

Having done this it looks to me like they are quite useful. So perhaps we should go ahead with putting them in a dl in a callout?

**Comment by pepelsbey** [comment]
Thank you for starting the discussion. I’m closing this now due to inactivity, but if you think that’s a mistake, feel free to reply. I’d also suggest filing a tracking issue in the [content repository](https://github.com/mdn/content/) to keep it more visible and take care of it someday. You’re also welcome to join the [MDN Discord](https://developer.mozilla.org/discord) if you’d prefer to chat with us more about it in sync.
